ts/codec-template.ts.j2

```diff
@@ -110,7 +110,12 @@ const RESPONSE_{{ to_upper_snake_case(param.name) }}_OFFSET = {% if loop.first %
 const EVENT_{{ to_upper_snake_case(event.name)}}_{{to_upper_snake_case(param.name)}}_OFFSET = {% if loop.first %}PARTITION_ID_OFFSET + BitsUtil.INT_SIZE_IN_BYTES{% else %}EVENT_{{ to_upper_snake_case(event.name)}}_{{ to_upper_snake_case(loop.previtem.name)}}_OFFSET + BitsUtil.{{loop.previtem.type.upper()}}_SIZE_IN_BYTES{% endif %};
     {% endfor %}
 {% endfor %}
-{% if method.response.params|length > 0 %}
+{#RESPONSE PARAMETERS#}
+{% set noResponseValue = method.response.params|length == 0 and response_new_params|length == 0 %}
+{% set singleResponseValue = method.response.params|length == 1 and response_new_params|length == 0 %}
+{% if noResponseValue %}
+{% elif singleResponseValue %}
+{% else %}
```

Comment:
Minor: Can we replace this part with something like `{% if not (noResponseValue or singleResponseValue) %} `?

Reply:
Simplified for both Node.js and Java templates in 658ca896698d705354c8af0434cd5b878b908c88